File: babydata/src/main/java/com/zero/babydata/room/BabyInfoDao.java
package com.zero.babydata.room;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Delete;
import androidx.room.Query;

import com.zero.babydata.entity.BabyInfo;

@Dao
public interface BabyInfoDao {

    @Insert
    void insertBabyInfo(BabyInfo babyInfo);

    @Query("SELECT * FROM Baby_Info WHERE babyId = :babyId LIMIT 1")
    BabyInfo getBabyInfo(int babyId);

    @Delete
    void deleteBabyInfo(BabyInfo babyInfo);

    @Query("UPDATE Baby_Info SET bloodType = :bloodType WHERE babyId = :babyId")
    void updateBloodType(int babyId, String bloodType);  // 更新血型
}

File: babydata/src/main/java/com/zero/babydata/room/BabyInfoDao.kt
package com.zero.babydata.room

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.zero.babydata.entity.BabyInfo

@Dao
interface BabyInfoDao {

    @Insert
    fun insertBabyInfo(babyInfo: BabyInfo)

    @Update
    fun updateBabyInfo(babyInfo: BabyInfo)

    @Delete
    fun deleteBabyInfo(babyInfo: BabyInfo)

    @Query("SELECT * FROM Baby_Info WHERE babyId = :babyId LIMIT 1")
    fun getBabyInfo(babyId: Int): BabyInfo?

    @Query("SELECT * FROM Baby_Info ORDER BY babyId DESC")
    fun getAllBabyInfo(): List<BabyInfo>

    @Query("UPDATE Baby_Info SET bloodType = :bloodType WHERE babyId = :babyId")
    fun updateBloodType(babyId: Int, bloodType: String?)
}
